components/sup/src/manager/mod.rs

```diff
@@ -669,7 +669,16 @@ impl Manager {
             Some(ref evg) => Some(events::EventsMgr::start(evg.clone())),
             None => None,
         };
-        signals::init();
+        // On Windows initializng the signal handler will create a ctrl+c handler for the
+        // process which will disable default windows ctrl+c behavior and allow us to
+        // handle via check_for_signal. However, if the supervsor is in a long running
+        // non-run hook, the below loop will not get to check_for_signal in a reasonable
+        // amount of time and the supervisor will not respond to ctrl+c. On Windows, we
+        // let the launcher catch ctrl+c and gracefully shut down services. ctrl+c should
+        // simply halt the supervisor
+        if cfg!(not(windows)) {
+            signals::init();
+        }
```

Comment:
@mwrock To make sure I've got the right understanding, once #5329 is done this could be reverted and still behave correctly, right?

I suppose this also means that there won't be any restarting of the Supervisor (with #5445) on Windows with this in place.

Just thinking out loud, could there be an alternative appropriate fix in `signals::init()` that could handle things in a cross-platform way?

Reply:
As far as I can tell, the only thing the `signals` implementation in our `core` crate does for windows is set up a ctrl-c handler, so https://github.com/habitat-sh/habitat/pull/5445 didn't add any functionality to that platform. I'll file an issue against the `core` repo to do more thorough handling on windows.

If we make this change (which I think is appropriate), we should also add a `not(windows)` conditions to the other references:

```rust
use hcore::os::signals::{self, SignalEvent};
```
on L56 and
```rust
            if let Some(SignalEvent::Passthrough(Signal::HUP)) = signals::check_for_signal() {
```
on L699

Currently we get lucky because [ATOMIC_BOOL_INIT](https://github.com/habitat-sh/core/blob/master/components/core/src/os/signals/windows.rs#L24) is `false`, but we shouldn't be calling `signals` functions if we haven't called `init`.

Reply:
I think ideally `signals::init` would be called on windows and we would need no platform branching. Then below we would check for both `SIGHUP`(signaled on Linux) and `Shutdown`(signaled on Windows) and do the same thing for both:`self::shutdown`.

We just can't do this until we get non-run hooks in their own threads. Actually Linux suffers from the same potential problem here that a long running init hook would delay the shutdown. Its just that Windows plans are much more likely to have long runnin init hooks and thus this is a more dire problem on that platform.

Reply:
but yeah, you are right that we should hide the above 2 lines for now on Windows. I'll add that.

Reply:
Filed https://github.com/habitat-sh/core/issues/59

Reply:
Maybe we should add an environment variable to toggle the signal catching behavior on or off if it can apply to either platform. Better to add that now (or just after this release if you think it's too much) than scramble later if it's affecting a production system.

Reply:
thats a great idea and I could rev a new windows-service package that sets that variable